# Section 1.9 Exercise 10

In [2]:
data(cars)
head(cars)

speed dist
1 4      2  
2 4     10  
3 7      4  
4 7     22  
5 8     16  
6 9     10

## Part A

In [10]:
# part a, fit a linear model of specified form, and then select the most appropriate model using both AIC and hypothesis testing
car.mod <- lm(dist ~ speed + I(speed^2), cars)

In [11]:
# first, model selection using AIC
car.mod2 <- lm(dist ~ speed, cars)
car.mod3 <- lm(dist ~ I(speed^2), cars)
car.mod4 <- lm(dist ~ 1, cars)
AIC(car.mod, car.mod2, car.mod3, car.mod4)

df AIC     
car.mod  4  418.7721
car.mod2 3  419.1569
car.mod3 3  416.9860
car.mod4 2  469.8024

**Model selection using AIC shows that car.mod3 (model containing only the speed^2 variable) is the most suitable model due to the fact that it has the lowest AIC out of the 4 models considered.**

In [16]:
# now, model selection using forwards model selection (hypothesis testing)
summary(lm(dist ~ 1, cars))


Call:
lm(formula = dist ~ 1, data = cars)

Residuals:
   Min     1Q Median     3Q    Max 
-40.98 -16.98  -6.98  13.02  77.02 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)   42.980      3.644   11.79 6.38e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 25.77 on 49 degrees of freedom


The model with the intercept only has a sufficiently low p-value, so I will continue adding predictor variables

In [17]:
summary(lm(dist ~ speed, cars))


Call:
lm(formula = dist ~ speed, data = cars)

Residuals:
    Min      1Q  Median      3Q     Max 
-29.069  -9.525  -2.272   9.215  43.201 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) -17.5791     6.7584  -2.601   0.0123 *  
speed         3.9324     0.4155   9.464 1.49e-12 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 15.38 on 48 degrees of freedom
Multiple R-squared:  0.6511,	Adjusted R-squared:  0.6438 
F-statistic: 89.57 on 1 and 48 DF,  p-value: 1.49e-12


**The model with the speed variable produces a sufficiently low p-value for the speed coefficient, so I will continue adding predictor variables**

In [18]:
summary(lm(dist ~ speed + I(speed^2), cars))


Call:
lm(formula = dist ~ speed + I(speed^2), data = cars)

Residuals:
    Min      1Q  Median      3Q     Max 
-28.720  -9.184  -3.188   4.628  45.152 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)
(Intercept)  2.47014   14.81716   0.167    0.868
speed        0.91329    2.03422   0.449    0.656
I(speed^2)   0.09996    0.06597   1.515    0.136

Residual standard error: 15.18 on 47 degrees of freedom
Multiple R-squared:  0.6673,	Adjusted R-squared:  0.6532 
F-statistic: 47.14 on 2 and 47 DF,  p-value: 5.852e-12


**The model with both the speed and speed^2 variables produces p-values greater than 0.05 for both variables, so I will omit this last model and choose the previous model with just the speed variable as the appropriate model.**

**However, it is clear that a model with only the speed^2 variable also produces a sufficiently small p-value, so for sake of consistency with the AIC model selection method, I will choose the model with just speed^2 as the appropriate model for the next steps of this exercise.**

In [20]:
summary(lm(dist ~ I(speed^2), cars))


Call:
lm(formula = dist ~ I(speed^2), data = cars)

Residuals:
    Min      1Q  Median      3Q     Max 
-28.448  -9.211  -3.594   5.076  45.862 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  8.86005    4.08633   2.168   0.0351 *  
I(speed^2)   0.12897    0.01319   9.781  5.2e-13 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 15.05 on 48 degrees of freedom
Multiple R-squared:  0.6659,	Adjusted R-squared:  0.6589 
F-statistic: 95.67 on 1 and 48 DF,  p-value: 5.2e-13


## Part B

In [32]:
# part b, from the selected model, estimate the avg time it takes a driver to apply the brakes
final.mod <- lm(dist/5280 ~ I(speed^2), cars)
time_pred <- predict(final.mod)
mean(time_pred)

[1] 0.008140152

**Using the selected model, the average time that it takes a driver to apply the brakes is 0.00814 seconds**

## Part C

**In the case of selecting between models with different polynomials, it is sensible to remove high powers of some continuous predictor and keep lower powers of that predictor. This is because the predictor of higher power and the predictor of lower power are directly collinear, or that that the model will do fine with just one of the variables (since the model will already be explained by that variable once). Removing the variable of higher power is essentially removing redundancy from the model and it is sensible to default to the more parismonious/conservative model.**